booking/src/main/java/org/dcsa/conformance/standards/booking/BookingScenarioListBuilder.java

```diff
@@ -78,101 +78,131 @@ private BookingScenarioListBuilder thenAllPathsFrom(BookingState bookingState) {
   private BookingScenarioListBuilder thenAllPathsFrom(
       BookingState bookingState, BookingState originalBookingState) {
     return switch (bookingState) {
-      case CANCELLED, COMPLETED, DECLINED, REJECTED  -> then(shipper_GetBooking(bookingState));
+      case COMPLETED  -> then(
+        shipper_GetBooking(bookingState).thenEither(
+         uc7_shipper_submitBookingAmendment(true)
+            .then(shipper_GetBooking(bookingState))
+        )
+      );
+      case DECLINED -> then(
+        shipper_GetBooking(bookingState).then(
+          uc7_shipper_submitBookingAmendment(true)
+            .then(shipper_GetBooking(bookingState))
+        )
+      );
+      case CANCELLED  -> then(
+        shipper_GetBooking(bookingState).thenEither(
+          uc3_shipper_submitUpdatedBookingRequest(true)
+            .then(shipper_GetBooking(bookingState))
+        )
```

Comment:
If `cancel` is not explicitly allowed on top of `cancel` then we should definitely have `cancel` here as well (for the return code check)